src/upgrade.py

```diff
@@ -90,6 +94,23 @@ def _on_postgresql_pebble_ready(self, event: WorkloadEvent) -> None:
             event.defer()
             return
 
+        if self.charm.unit.is_leader():
+            if not self.charm._patroni.primary_endpoint_ready:
+                logger.debug(
+                    "Deferring on_pebble_ready: current unit is leader but primary endpoint is not ready yet"
+                )
+                event.defer()
+                return
+            self._set_up_new_credentials_for_legacy()
+
+        try:
+            self.charm.unit.set_workload_version(
+                self.charm._patroni.rock_postgresql_version or "unset"
+            )
+        except PostgreSQLGetPostgreSQLVersionError:
+            # Don't fail on this, just log it.
+            logger.warning("Failed to get PostgreSQL version")
+
```

Comment:
Wouldn't the version be set anyway by the pebble_ready in `charm.py`?

Reply:
Good catch. I fixed it on https://github.com/canonical/postgresql-k8s-operator/pull/210/commits/7c8bed563bf4ad55fb64ed316cf50ce26c1069f6.